src/pages/iv-opening-cost/index.js

```diff
@@ -0,0 +1,148 @@
+import { useContext } from 'react'
+import toast from 'react-hot-toast'
+import Table from 'src/components/Shared/Table'
+import GridToolbar from 'src/components/Shared/GridToolbar'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { useResourceQuery } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Fixed } from 'src/components/Shared/Layouts/Fixed'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { useWindow } from 'src/windows'
+import { ControlContext } from 'src/providers/ControlContext'
+import { InventoryRepository } from 'src/repositories/InventoryRepository'
+import OpeningCostForm from './Forms/OpeningCostForm'
+import RPBGridToolbar from 'src/components/Shared/RPBGridToolbar'
+
+const OpeningCost = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50, params } = options
+
+    const response = await getRequest({
+      extension: InventoryRepository.OpeningCost.qry,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&filter=&_params=${params || ''}&_sortBy=year desc`
+    })
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  async function fetchWithFilter({ filters, pagination }) {
+    return fetchGridData({ _startAt: pagination._startAt || 0, params: filters?.params })
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    refetch,
+    access,
+    invalidate,
+    filterBy
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+    endpointId: InventoryRepository.OpeningCost.qry,
+    datasetId: ResourceIds.InventoryOpeningCosts,
+    filter: {
+      filterFn: fetchWithFilter
+    }
+  })
+
+  const columns = [
+    {
+      field: 'year',
+      headerName: _labels.fiscalYear,
+      flex: 1
+    },
+    {
+      field: 'sku',
+      headerName: _labels.sku,
+      flex: 1
+    },
+    {
+      field: 'itemName',
+      headerName: _labels.item,
+      flex: 1
+    },
+    {
+      field: 'avgCost',
+      headerName: _labels.cost,
+      flex: 1,
+      type: 'number'
+    }
+  ]
+
+  const add = () => {
+    openForm()
+  }
+
+  const edit = obj => {
+    openForm(obj)
+  }
+
+  function openForm(record) {
+    stack({
+      Component: OpeningCostForm,
+      props: {
+        labels: _labels,
+        recordId: record
```

Comment:
you are passing record and recordId, can u pass only record and inside the form add const recordId = String(record.year) + String(record.itemId)  || null